### PR TITLE
Improve `strict_loading` documentation [ci skip]

### PR DESCRIPTION
**Pull request by ansonhoyt**
* Update ActiveRecord::StrictLoadingViolationError API docs to
  reference the guide to help beginners learn how to resolve the error.
* Update Active Record Query Interface guide to cover configuration and
  the association option.

### Motivation / Background

This Pull Request has been created to improve the docs and guides.

### Detail

This Pull Request changes:

1. ActiveRecord::StrictLoadingViolationError API docs to link to the guide.
2. Active Record Query Interface guide to cover other facets of `strict_loading`.

### Additional information

Explains the association option, highlighted in Chris Oliver's excellent Rails World talk [Powerful Rails Features You Might Not Know](https://youtu.be/iPCqwZ9Ouc4?si=oFrWz5Db6zZYTYUN&t=134).

Continues the nice additions in https://github.com/rails/rails/pull/49329

I was inspired by @zzak's comment on https://github.com/rails/rails/pull/49741

### Checklist

Before submitting the PR make sure the following are checked:

* [x] This Pull Request is related to one change. Changes that are unrelated should be opened in separate PRs.
* [x] Commit message has a detailed description of what changed and why. If this PR fixes a related issue include it in the commit message. Ex: `[Fix #issue-number]`
* [ ] Tests are added or updated if you fix a bug or add a feature.
* [ ] CHANGELOG files are updated for the changed libraries if there is a behavior change or additional feature. Minor bug fixes and documentation changes should not be included.

### CODE REVIEW

File: activerecord/lib/active_record/errors.rb [hunk 1]
```diff
@@ -373,6 +373,14 @@ class ReadOnlyRecord < ActiveRecordError
...
 
   # Raised on attempt to lazily load records that are marked as strict loading.
+  # This avoids <tt>N+1</tt> queries.
+  #
+  # You can resolve this error by eager loading marked records before accessing
+  # them.  The
+  # {Eager Loading Associations}[https://guides.rubyonrails.org/active_record_querying.html#eager-loading-associations]
+  # guide covers solutions, such as using
+  # {ActiveRecord::Base.includes}[rdoc-ref:QueryMethods#includes] or setting
+  # {strict_loading}[rdoc-ref:QueryMethods#strict_loading].
```
Comment (eileencodes) [inline activerecord/lib/active_record/errors.rb:383]:
```suggestion
  # {ActiveRecord::Base.includes}[rdoc-ref:QueryMethods#includes].
```

If someone turned it on it doesn't make sense to document changing the setting IMO.

File: guides/source/active_record_querying.md [hunk 2]
```diff
@@ -1763,7 +1763,14 @@ user.address.city # raises an ActiveRecord::StrictLoadingViolationError
 user.comments.to_a # raises an ActiveRecord::StrictLoadingViolationError
 ```
 
+To enable for all relations, change the
+[`config.active_record.strict_loading_by_default`][] flag to `true`.
+To send violations to the logger instead, change
```
Comment (eileencodes) [inline guides/source/active_record_querying.md:1768]:
```suggestion

To send violations to the logger instead, change
```

**Review by eileencodes — CHANGES_REQUESTED** [review]

**Review by akhilgkrishnan — APPROVED** [review]

**Comment by eileencodes** [comment]
Thanks, can you squash your commits into one commit?

**Comment by ansonhoyt** [comment]
@eileencodes Done! Thanks for bearing with my ~~Codespaces~~ learning curve.

P.S. I enjoyed your talk [The Magic of Rails](https://www.youtube.com/watch?v=nvuPisDQ1hI). It piqued my curiosity to explore Active Storage code as I write a small API client.

**Review by eileencodes — APPROVED** [review]

**Comment by zzak** [comment]
Thanks for working on this! Good additions. 👏 

> The API Documentation Guidelines' [Links](https://edgeguides.rubyonrails.org/api_documentation_guidelines.html#links) section doesn't cover when or how we should link to the guides from API docs.
> 
> I think it would help me if I got this error, but I couldn't find any existing links from API docs to Guides, so let me know if I did a bad thing 😬

There is a tiny blurb about it here, if you mean what I think you mean:
https://edgeguides.rubyonrails.org/ruby_on_rails_guides_guidelines.html#linking-to-the-api

In short, `api.` and `guides.` get turned into `edge*` during generation. So we don't have to type "edge" unless we really mean it.

**Pull request merged**